alibi_detect/utils/tensorflow/distance.py

```diff
@@ -20,14 +26,44 @@ def squared_pairwise_distance(x: tf.Tensor, y: tf.Tensor, a_min: float = 1e-30,
         Lower bound to clip distance values.
     a_max
         Upper bound to clip distance values.
-
+    pairwise
```

Comment:
Isn't it a bit unclear to have a function named `squared_pairwise_distance` that optionally computes non-pairwise distances? Perhaps `squared_pairwise_distance` should be renamed `squared_distance`? Or the `pairwise=False` functionality separated out into a separate distance function?

Reply:
I was thinking the same.  It was previously a separate function and @arnaudvl was suggesting making the repeated parts minimal. Guess changing the function name across all related methods would be preferable.